src/repository/DataSource.ts

```diff
@@ -14,16 +14,20 @@ export abstract class DataSource {
   }
 
   abstract getTokens();
+  abstract getBlockNumberByDate(date: number, closest: string);
   abstract getTokensByAddress(address: string);
   abstract getRbtcBalanceByAddress(address: string);
-  abstract getEventsByAddress(address: string, limit?: string);
+  abstract getEventsByAddress(address: string, limit?: string, startBlock?: number, endBlock?: number);
   abstract getTransaction(hash: string);
-  abstract getInternalTransactionByAddress(address: string, limit?: string);
+  abstract getInternalTransactionByAddress(address: string, limit?: string, startBlock?: number, endBlock?: number);
   abstract getTransactionsByAddress(address:string,
     limit?: string,
     prev?: string,
     next?: string,
-    blockNumber?: string);
+    blockNumber?: string,
+    startDate?: number,
```

Comment:
If startDate is expected to be unix, then rename it to unixStartDate
Also the same for endDate

You can also rename it to something that you think could have even more meaning, such as epochStartDate or likewise, up to you 😁

Reply:
Same as above